modules/shared/src/main/scala/io/constellationnetwork/currency/http/Decoders.scala

```diff
@@ -0,0 +1,42 @@
+package io.constellationnetwork.currency.http
+
+import cats.data.NonEmptyList
+import cats.effect.Async
+
+import io.constellationnetwork.currency.dataApplication.DataTransaction.DataTransactions
+import io.constellationnetwork.currency.dataApplication.{DataTransaction, DataUpdate, FeeTransaction}
+import io.constellationnetwork.security.signature.Signed
+
+import derevo.circe.magnolia.decoder
+import derevo.derive
+import io.circe.Decoder
+import org.http4s.circe.jsonOf
+import org.http4s.{DecodeResult, EntityDecoder, InvalidMessageBodyFailure}
+
+object Decoders {
+
+  def feeTransactionRequestDecoder[F[_]: Async, D <: DataUpdate](
+    implicit dataUpdateDecoder: Decoder[D],
+    feeTransactionDecoder: Decoder[FeeTransaction]
+  ): EntityDecoder[F, DataTransactions] = {
+    implicit val signedDataUpdateDecoder: Decoder[Signed[D]] = Signed.decoder[D]
+    implicit val signedFeeTransactionDecoder: Decoder[Signed[FeeTransaction]] = Signed.decoder[FeeTransaction](feeTransactionDecoder)
```

Comment:
I think I've already mentioned it too, you just need a decoder for `A` not for `Signed[A]`. So just `dataUpdateDecoder` and `feeTransactionDecoder`. Decoder for `Signed` should be automatically derived and if you know how to decode `A`, the compiler would know how to decode `Signed[A]`.

Reply:
👍🏼